src/system/time.rs

```diff
@@ -124,6 +124,15 @@ impl Sub<Duration> for Duration {
     }
 }
 
+impl From<Duration> for std::time::Duration {
+    fn from(dur: Duration) -> std::time::Duration {
+        std::time::Duration::new(
+            dur.secs.try_into().unwrap_or(0),
```

Comment:
I think this is a bit suspect, but it's a consequence of our choice (which I forgot why it was made exactly) to have our own Duration object. In any case I've analyzed the code that the `unwrap_or`s are never reached.

I'll spin that off into its own issue as it falls outside the scope of this PR.

Reply:
See #1170